src/app_service/service/chat/service.nim

```diff
@@ -187,126 +177,62 @@ QtObject:
           self.chats[chat.id] = chat
 
-  proc onAsyncGetChannelGroupsResponse*(self: Service, response: string) {.slot.} =
+  proc onAsyncGetActiveChatsResponse*(self: Service, response: string) {.slot.} =
     try:
       let rpcResponseObj = response.parseJson
 
       if (rpcResponseObj{"error"}.kind != JNull and rpcResponseObj{"error"}.getStr != ""):
         raise newException(CatchableError, rpcResponseObj{"error"}.getStr)
 
-      if(rpcResponseObj["channelGroups"].kind == JNull):
-        raise newException(RpcException, "No channel groups returned")
+      if(rpcResponseObj["chats"].kind == JNull):
+        raise newException(RpcException, "No chats returned")
 
-      self.hydrateChannelGroups(rpcResponseObj["channelGroups"])
-      self.events.emit(SIGNAL_CHANNEL_GROUPS_LOADED, ChannelGroupsArgs(channelGroups: self.getChannelGroups()))
+      self.hydrateChats(rpcResponseObj["chats"])
+      self.events.emit(SIGNAL_ACTIVE_CHATS_LOADED, Args())
     except Exception as e:
       let errDesription = e.msg
-      error "error get channel groups: ", errDesription
-      self.events.emit(SIGNAL_CHANNEL_GROUPS_LOADING_FAILED, Args())
+      error "error get active chats: ", errDesription
+      self.events.emit(SIGNAL_CHATS_LOADING_FAILED, Args())
 
   proc init*(self: Service) =
     self.doConnect()
 
-    self.asyncGetChannelGroups()
+    self.asyncGetActiveChat()
 
   proc hasChannel*(self: Service, chatId: string): bool =
     self.chats.hasKey(chatId)
 
-
-  proc getChatIndex*(self: Service, channelGroupId, chatId: string): int =
-    var i = 0
-
-    if not self.channelGroups.contains(channelGroupId):
-      warn "unknown channel group", channelGroupId
-      return -1
-
-    for chat in self.channelGroups[channelGroupId].chats:
-      if (chat.id == chatId):
-        return i
-      i.inc()
-    return -1
-
-  proc chatsWithCategoryHaveUnreadMessages*(self: Service, communityId: string, categoryId: string): bool =
-    if communityId == "" or categoryId == "":
-      return false
-
-    if not self.channelGroups.contains(communityId):
-      warn "unknown community", communityId
-      return false
-
-    for chat in self.channelGroups[communityId].chats:
-      if chat.categoryId != categoryId:
-        continue
-      if (not chat.muted and chat.unviewedMessagesCount > 0) or chat.unviewedMentionsCount > 0:
-        return true
-    return false
-
   proc sectionUnreadMessagesAndMentionsCount*(self: Service, communityId: string):
       tuple[unviewedMessagesCount: int, unviewedMentionsCount: int] =
-    if communityId == "":
-      return
-
-    if not self.channelGroups.contains(communityId):
-      warn "unknown community", communityId
-      return
 
     result.unviewedMentionsCount = 0
     result.unviewedMessagesCount = 0
 
-    for chat in self.channelGroups[communityId].chats:
+    let myPubKey = singletonInstance.userProfile.getPubKey()
+    var commId = communityId
+    if communityId == myPubKey:
+      commId = ""
```

Comment:
We don,t want to return. This function is used by the personal chat section too. I'll rename the param to `sectionId` id, because I agree that using `communityId` is confusing.
I'll also add a comment to explain the conversion